### PR TITLE
Codebuild

### PR DESCRIPTION
**Pull request by amywieliczka**
### Infrastructure Updates:
- Allow Codebuild Execution Role to access ECR (required to push up built image) and output ECR URI
- Update app-servers.yaml config to use the built image, add env vars, update container port, entrypoint, command
- Update Codebuild build spec *not* to use buildx - we were hitting docker rate limiting issues in trying to use buildx
- Add a task role policy with access to s3, attach it to the task definition for the cinco ctrl application
- **Not included in PR, manually done in AWS Console**: Turned off "block all public access" to the `awieliczka-test-bucket` (used in the pad-dev acct), and added the following bucket-level permissions policy:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Allow Public Read of /static",
            "Effect": "Allow",
            "Principal": "*",
            "Action": "s3:GetObject",
            "Resource": [
                "arn:aws:s3:::awieliczka-test-bucket/static/*",
                "arn:aws:s3:::awieliczka-test-bucket/media/*"
            ]
        }
    ]
}
```

### Cinco Ctrl Build Environment Updates:
- Remove Redis
- Remove Traefik

### Production Settings Updates:
- Caching: configured local memcache (instead of Redis) as cache of choice
- Sentry: commented out for now
- S3 Backend: removed AWS access key env vars - rely on the task role policy instead; remove default_acl from config - the bucket has permissions set in the bucket policy, not through object ACLs, per AWS recommendation. 
- Allowed Hosts: added * wildcard to - we'll probably want to lock this down later once we have persistent DNS. I'm curious if we'll need to add the ALB's public IP address at that time - I was getting some errors to this effect. May alternately need to look into middleware and the forwarded for header. Will definitely need to get private IP of the container in order for the ECS task to pass its own health check. 
- Debug: turned on for now
- Email: swapped out console mail backend for anymail backend (commented out)
- Security: commented out a variety of CSRF-HTTPS-related configuration settings

### Infrastructure Scripting:
- Created an `ecs_manage.py` script that passes command line arguments through to `python manage.py` on the ECS container - reuses the same task definition as the application (and all the same roles/policies/security/etc) - calls run task with a container command override. Does *not* put the task in the same ECS service, so the task is not attached to the load balancer and is protected from serving web application traffic. 
- Can be used to run from your local machine (with an AWS access key in the local environment): `python ecs_manage.py createsuperuser --no-input --email amywieliczka@gmail.com`, or `python ecs_manage.py collectstatic --no-input`, or `python ecs_manage.py migrate`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,11 @@
 Install pyenv, and it should automagically use the python version specified in `.python-version`
 
 Tooling is listed in dev-requirements.txt
+
+## Run Development Server
+
+`docker compose -f docker-compose.local.yml up`
```
Comment (bibliotechy) [inline README.md:9]: Don't both of these need to be run in the `admin/cincoctrl/` directory? Which has it's own slightly different instructions for docker setup. Is there a reason to have those in to places?

File: admin/cincoctrl/config/settings/production.py [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+# import sentry_sdk
+# from sentry_sdk.integrations.django import DjangoIntegration
+# from sentry_sdk.integrations.logging import LoggingIntegration
 
 from .base import *  # noqa: F403
 from .base import DATABASES
 from .base import INSTALLED_APPS
 from .base import env
 
+DEBUG = True
```
Comment (bibliotechy) [inline admin/cincoctrl/config/settings/production.py:13]: I don't think we want this as always `True` in the production settings. Is this just for development?

**Review by bibliotechy — APPROVED** [review]
Looks good. Two small questions.

I like the approach of a built in shell script to run migrations (and other changes) because I can easily see it being part of an automated piepline for deployments, but also able to be from a developers laptop locally in a pinch. I can see that being part of what calls django custom commands from Airflow in the future.

But once we have this in prod with a bunch of prod data, I think adding some guardrails might be useful as safeguards against accidental data deletion, updates, etc. Maybe it's fine, but easy to tab up through your history and accidentally run a non-idempotent script locally.

**Review by everreau — APPROVED** [review]

File: admin/cincoctrl/config/settings/production.py (hunk 2 above)
Comment (amywieliczka) [reply admin/cincoctrl/config/settings/production.py:13]: Yeah, I conflated "deployed" with "production" - I think some follow up work involves creating a "stage.py" settings file and related resources. I think these three commits include things we'll want to roll back for production: 

https://github.com/ucldc/cinco/commit/4ecb929d1a6215b057b377554979db51ecb8f02f
https://github.com/ucldc/cinco/commit/580f08c86142f132381b2e6faff3a8753d0a4c24
https://github.com/ucldc/cinco/commit/0530838c9f3514b1928a7de4b96c1176235d66c1

File: README.md (hunk 1 above)
Comment (amywieliczka) [reply README.md:9]: This should get moved into the README located at `admin/cincoctrl/` - this is in fact how to run the development server though, even though it doesn't exist in the README there. Will address is upcoming reorg PR.

**Pull request merged**